### DISCUSSION TITLE
Informations about annotation written by SumatraPDF

### DISCUSSION THREAD

**Discussion by axeldaguerre** [sumatrapdfreader/sumatrapdf#4317] [Q&A]
SumatraPDF seems to not properly write the new line /r/n properly instead it's written as '\\' + 'r', it should be ''\r'.
As SumatraPDF is certainly parsing the annotation (highlight or other) by itself it doesn't care. 
But when you are parsing yourself the PDF saved with SumatraPDF, parsing is more difficult, I have to do :
```C
if(*byte == '\\' && byte[1] == 'r') || 
   *byte == '\\' && byte[1] == 'n')
  {}
 ```
 Which quite ugly. Instead of 
 ```C
if(*byte == '\r'  || 
    *byte == '\n' )
  {}
 ```
 If someone think it's bad and care (I don't), you may be aware of that and maybe fix it.
 To be more specific, using annotation tool: 
 ```txt
3271 0 obj
<</Type/Annot/Subtype/Highlight/C[1 1 0]/P 170 0 R/F 4/M(D:20230324055415Z)/T(axeld)/Rect[128.4072 439.3125 414.5928 484.6875]/AP<</N 3275 0 R>>/QuadPoints[144 484 356 484 144 473 356 473 131 467 285 467 131 457 285 457 131 451 412 451 131 440 412 440]/Contents(Converting an improper fraction\r\nto a mixed number)>>
endobj
 ```
 
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/51819886/c092954e-6e7b-4b85-9bac-0ab8101b266f)

Knowing that SumatraPDF uses mupdf library but not in which extend, maybe that easy to solve.
Axel

**Comment by GitHubRulesOK** [comment]
There is an issue open https://github.com/sumatrapdfreader/sumatrapdf/issues/1694#issuecomment-864689978 about feeding lines since it is SumatraPDF that does it different to MuPDF and Edge but not all are the same see the comparisons on the right (apart from my omission in the last "mixed") Tracker uses \r and wraps the second line. But for cross platform compatability between \r \n and Windows \r\n Adobe suggest just \n.

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/119ba433-1bf5-406d-9f5f-766ff7f05599)

```
6 0 obj<</AP<</N 18 0 R>>/Border[0 0 0]/DA(/Helv 10.5 Tf 0 Tc 11.76 TL 0 0.470588 0.831373 rg )/F 4/NM(c094e820-ff51-46bc-ba9c69e889179087)/Rect[434.403 349.5 587.457 385.5]/Rotate 0/Subtype/FreeText/Type/Annot/PDFIUM_HasGeneratedAP true

/Contents(MS Edge\nConverting an improper fraction\nto a mixed number)

>>endobj
7 0 obj<</AP<</N 19 0 R>>/BS<</Type/Border/W 0>>/DA(/Helv 11 Tf 0 0 0 rg)/F 4/M(D:20240612142419Z)/NM(2f74206f-aaa2-4b9f-8df62efcc5d12376)/P 5 0 R/Rect[427.56446 227.90164 627.56448 327.90165]/Subtype/FreeText/T(lez)/Type/Annot

/Contents(SumatraPDF\r\nConverting an improper fraction\r\nto a mixed number)

>>endobj
8 0 obj<</AP<</N 20 0 R>>/BS<</W 0>>/C[1 1 1]/CreationDate(D:20240612162420+01'00')/DA(1 0 0 rg /F0 10 Tf)/DS(font-family:Arial;font-size:10pt;color:#000000;)/F 4/IT/FreeText/M(D:20240612162649+01'00')/NM(e539d677-eb6e-40b8-94c596663f81c17a)/P 5 0 R/Rect[431.70506 181.16469 580.5006 230.83469]/Subj(Text Box)/Subtype/FreeText/T(lez)/Type/Annot

/Contents(PDF-XChange\rConverting an improper fraction to a mixed number)

>>endobj
32 0 obj <</Type/Annot/Subtype/FreeText/Rect[426.57579 91.5 593.45529 171.5]/BS<</W 0>>/DA(/Helv 11 Tf 0 0 1 rg)/P 5 0 R/F 4/M(D:20240612152935Z)/T(lez)/AP<</N 34 0 R>>

/Contents(MuPDF\nConverting an improper fraction\nto a number)

>>endobj
```

**Reply by axeldaguerre** [reply]
I think you misunderstood what I was saying (maybe that's my bad English's fault). The issue I had may have been because I wasn't decoding the text correctly because it treated '\r' incorrectly. Not sure about that yet, but it's a possibility. I am not looking for an answer or an issue creation, though. It was just an information I found and could help the maintainers of the project.
I am not expert in Github so maybe it wasn't the place to post it.
Axel

**Reply by GitHubRulesOK** [reply]
Sumatra PDF is Windows only and PDF is BYTES so there is no \r or \n just octal \015 (^M) and \013 (^J) = Move carriage to start & Jump paper up a line.

DOS ENTER key is DESCRIBED in text literals as `\r\n` so Sumatra does not filter the 2 keycodes as if one 
Mac enter = \r Linux enter = \n so PDF treats \r\n as one linefeed in its handling of instructions like in a text editor
but they count as 2 linefeeds within a comment = one Mac <0D> one Linux <0A> !!

Adobe say when it is important like in the xref index then only use the Linux format (0A =\n )
\r is to be discouraged in Mac editing of PDF (thus can mean Mac files misbehave!)
